contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2679,6 +2679,16 @@ rewriteBatchLevelStatement(
 				auto commas = cctx->COMMA();
 				GetTokenFunc<TSqlParser::Function_optionContext*> getToken = [](TSqlParser::Function_optionContext* o) { return o->SCHEMABINDING(); };
 				bool all_removed = removeTokenFromOptionList(expr, options, commas, ctx, getToken);
+				/* Check if execute_as_clause is present, and if is present check for caller clause */
+				bool execute_as_clause_check = false;
+				for (auto option : options)
+					if (option->execute_as_clause())
+						execute_as_clause_check = true;
+				if(execute_as_clause_check)
+				{
+					getToken = [](TSqlParser::Function_optionContext* o) { return o->execute_as_clause()->CALLER(); };
+					all_removed = removeTokenFromOptionList(expr, options, commas, ctx, getToken);
```

Comment:
in "removeTokenFromOptionList"
```
bool
removeTokenFromOptionList(PLtsql_expr *expr, std::vector<T>& options, std::vector<antlr4::tree::TerminalNode *>& commas, antlr4::ParserRuleContext *ctx, GetTokenFunc<T> getTokenFunc)
{
	Assert(options.size() -1 == commas.size());

	bool all_removed = true;
	bool comma_carry_over = false;

	for (size_t i=0; i<options.size(); ++i)
	{
		if (getTokenFunc(options[i]))
```

getTokenFunc(options[i]) is called for all of the options. Now we have passed o->execute_as_clause()->CALLER() as getTokenFunc but not all options (o) will have a NOT NULL execute_as_clause variable. Hence, any option other than "execute as clause" option will lead to seg fault (o->NULL->CALLER())